io-engine/src/lib.rs

```diff
@@ -45,3 +47,9 @@ pub extern "C" fn cps_init() {
     bdev::nexus::register_module();
     bdev::null_ng::register();
 }
+
+/// Event trait definition for creating events.
+pub(crate) trait Event {
+    /// Create event message.
+    fn event(&self, event_action: Option<EventAction>) -> EventMessage;
```

Comment:
Why is action optional?

Reply:
I've allowed for event_action to be optional in case it isn't required any particular event generation. But there are no events with event_action of none for now. I can remove the optional if necessary.

Reply:
I see, it's fine then I was just curious.